src/index.ts

```diff
@@ -0,0 +1,185 @@
+    const npxCMD = command.startsWith("wrangler")
+      ? `npx ${command}`
+      : command;
+
+    info(`🚀 Executing command: ${npxCMD}`);
+
+    execSync(npxCMD, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+    });
+  }
+  endGroup();
+}
+
+async function uploadSecrets() {
+  startGroup("🔑 Uploading Secrets");
+  const secrets: string[] | string = config["secrets"];
+  if (!secrets.length) {
+    warning(`📌 No secrets were provided, skipping upload.`);
+    return;
+  }
+  const npxCMD =
+    process.env.RUNNER_OS === "Windows" ? "npx.cmd exec" : "npx";
+  const environment = config["ENVIRONMENT"];
+
+  const getSecret = (secret: string) => process.env[secret] ?? "";
+  const secretObj = secrets.reduce((acc: any, secret: string) => {
+    acc[secret] = getSecret(secret);
+    return acc;
+  }, {});
+
+  const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+  const secretCmd = `${npxCMD} wrangler secret:bulk ${JSON.stringify(
+    secretObj
+  )}${environmentSuffix}`;
+
+  try {
+    spawnSync(secretCmd, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+      stdio: "ignore"
+    });
+    info(`✅ Uploaded secrets`);
+  } catch (error) {
+    setFailed(`🚨 Failed to upload secrets`);
+  }
+
+  endGroup();
+}
+
+async function genericCommand() {
+  startGroup("🚀 Executing Generic Command");
+  const commands = config["COMMANDS"];
+  if (!commands.length) {
+    warning(`📌 No generic commands were provided, skipping execution.`);
+    return;
+  }
+  const wranglerVersion = config["WRANGLER_VERSION"];
+  const environment = config["ENVIRONMENT"];
+  const vars = config["VARS"];
+  const workingDirectory = config["workingDirectory"];
+
+  if (commands.length === 0) {
+    const deployCommand =
+      wranglerVersion === "latest" || wranglerVersion.startsWith("3")
+        ? "deploy"
+        : "publish";
+
+    warning(`🚨 No commands were provided, falling back to '${deployCommand}'`);
+
+    const envVarArray = vars.map((envVar: string) => {
+      if (process.env[envVar] && process.env[envVar]?.length !== 0) {
+        return `${envVar}:${process.env[envVar]!}`;
+      } else {
+        setFailed(`🚨 ${envVar} not found in variables.`);
+      }
+    });
+
+    const envVarArg: string =
+      envVarArray.length > 0 ? `--var ${envVarArray.join(" ").trim()}` : "";
+
+    if (environment.length === 0) {
+      execSync(`npx wrangler ${deployCommand} ${envVarArg}`.trim(), {
+        cwd: workingDirectory,
+        env: process.env,
+      });
+    } else {
+      execSync(
+        `npx wrangler ${deployCommand} --env ${environment} ${envVarArg}`.trim(),
+        { cwd: workingDirectory, env: process.env }
+      );
+    }
+  } else {
+    if (environment.length === 0) {
+      warning(
+        `🚨 An environment as been specified adding '--env ${environment}' is required in the command.`
+      );
+    }
+
+    return execCommands([`npx wrangler ${commands}`]);
```

Comment:
Should this be:

```suggestion
    return execCommands(commands.map(cmd => `npx wrangler ${cmd}`]);
```

I'm not sure how multiline input will be executed your way?